api/folders/controllers/folders.js

```diff
@@ -4,28 +4,51 @@ const { sanitizeEntity } = require("strapi-utils");
 
 module.exports = {
   async tree(ctx) {
-    let entities = await strapi.services.folders.find();
-    entities = entities.map((entity) =>
+    let foldersEntities = await strapi.services.folders.find();
+
+    foldersEntities = foldersEntities.map((entity) =>
       sanitizeEntity(entity, { model: strapi.models.folders })
     );
+
+    const { tag_id_in } = ctx.query;
+    let bookmarksFiltered = [];
+    if (tag_id_in) {
+      const tags = eval(tag_id_in);
+      let tagsEntities = await strapi.services.tags.find({ id_in: tags });
+      tagsEntities = tagsEntities.map((entity) =>
+        sanitizeEntity(entity, { model: strapi.models.tags })
+      );
+      tagsEntities.forEach((tagEntity) => {
+        tagEntity.bookmarks.forEach((bookmark) => {
+          bookmarksFiltered[bookmark.id] = bookmark;
+        });
+      });
+    }
+    const bookmarksFilteredId = bookmarksFiltered.map((bookmark) => {
+      return bookmark.id;
+    });
+    foldersEntities = foldersEntities.map((entity) => {
+      entity.bookmarks = entity.bookmarks.filter(
+        (bookmark) => !bookmarksFilteredId.includes(bookmark.id)
+      );
```

Comment:
Open question : are we keeping empty folders ? (I think yes with this code)

Reply:
Yes indeed; I'm questionning myself, if this is a UI choice to remove them or a server need

Reply:
Ok; so I added a no_empty_folders param ;) Check last commit

Reply:
It resolves my internal conflict on UI or not lol